toolkit/pythonit_toolkit/api/client.py

```diff
@@ -0,0 +1,54 @@
+import os
+from typing import Any, Dict, Optional
+
+import httpx
+import jwt
+from pythonit_toolkit.headers import SERVICE_JWT_HEADER
+
+INTERNAL_JWT_SECRET = os.getenv("INTERNAL_JWT_SECRET", "a-very-secret-secret")
+
+
+class Client:
+    url = "/internal-api"
+
+    async def execute(
+        self,
+        query: str,
+        variables: Optional[Dict[str, Any]] = None,
+    ):
+        token = jwt.encode({"some": "payload"}, INTERNAL_JWT_SECRET, algorithm="HS256")
+
+        async with httpx.AsyncClient() as client:
+            response = await client.post(
+                self.url,
+                data={"query": query, "variables": variables},
+                headers={
+                    SERVICE_JWT_HEADER: token,
+                },
+            )
+
+            data = await response.json()
+            return data
+
+    def users(self):
```

Comment:
I don't think the client should have queries in it. It ideally should only expose your `execute` and let the various services send the query

Reply:
Yeah I know is not ideal to return all the attributes any time. I thought we can maybe return an object instead of a dictionary if we know what we are looking for...